ts/Data/DataSeriesConverter.ts

```diff
@@ -0,0 +1,267 @@
+/* *
+ *
+ *  Data Layer
+ *
+ *  (c) 2012-2020 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type DataEventEmitter from './DataEventEmitter';
+import type LineSeries from '../Series/Line/LineSeries';
+import type PointOptions from '../Core/Series/PointOptions';
+import type SeriesOptions from '../Core/Series/SeriesOptions';
+import DataTable from './DataTable.js';
+import DataTableRow from './DataTableRow.js';
+import U from '../Core/Utilities.js';
+const {
+    defined,
+    uniqueKey
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Class to convert DataTable to Highcharts series data.
+ */
+class DataSeriesConverter {
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+    public constructor(table: DataTable = new DataTable(), options: DataSeriesConverter.Options) {
+        this.table = table;
+        this.options = options;
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+    public table: DataTable;
+    public options: DataSeriesConverter.Options;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    getSeriesData(columnIndex: number): Array<PointOptions> {
+        const table = this.table,
+            options = this.options || {},
+            dataOptions = [],
+            seriesTypeData = this.getSeriesTypeData(options.type || 'line');
+
+        let row,
+            column;
+
+        for (let i = 0, iEnd = table.getRowCount(); i < iEnd; i++) {
+
+            row = table.getRow(i);
+
+            if (row) {
+                column = row.getCell(row.getCellNames()[columnIndex]);
+
+                if (typeof column === 'number') {
+                    dataOptions.push(
+                        seriesTypeData(column)
+                    );
+                }
+            }
+        }
+
+        return dataOptions;
+    }
+
+    getSeriesTypeData(type: string): Function {
+        let fcName: Function;
+
+        switch (type) {
+        case 'line':
+            fcName = this.getLinePoint;
+            break;
+        case 'pie':
+            fcName = this.getPiePoint;
+            break;
+        case 'range':
+            fcName = this.getRangePoint;
+            break;
+        default:
+            fcName = this.getLinePoint;
+            break;
+        }
+
+        return fcName;
+    }
+
+    getLinePoint(column: number): PointOptions {
+        return {
+            y: column
+        };
+    }
+
+    getPiePoint(column: number): PointOptions {
+        return {
+            y: column
+        };
+    }
+
+    getRangePoint(column: number): PointOptions {
+        return {
+            y: column
+        };
+    }
```

Comment:
Why are these separate functions?

Reply:
The idea is: distinguish the point type. I.e getRangePoint will return `y` and `yLow` values.  

I agree that the `getPiePoint` is unnecessary.